Material.Blazor/Components/Dialog/MBDialog.razor

```diff
@@ -1,19 +1,19 @@
 ﻿@namespace Material.Blazor
 @inherits ComponentFoundation
 
+<div @ref="DialogElem"
+     class="mdc-dialog @ActiveConditionalClasses @(@class)"
+     style="@style"
+     id="@id"
+     @attributes="@AttributesToSplat()"
+     aria-labelledby="@(HasTitle ? titleId : (HasCustomHeader ? headerId : null))"
+     aria-describedby="@(HasBody ? bodyId : null)"
+     aria-modal="true">
 
-@if (IsOpen)
-{
-    <div @key="@Key"
-         @ref="DialogElem"
-         class="mdc-dialog @ActiveConditionalClasses @(@class)"
-         style="@style"
-         id="@id"
-         @attributes="@MyAttributes"
-         aria-modal="true">
-
-        <CascadingValue Value="@this" IsFixed="true">
-            <div class="mdc-dialog__container">
+    <CascadingValue Value="@this" IsFixed="true">
+        <div class="mdc-dialog__container">
+            @if (IsOpen)
```

Comment:
Only the dialog content is now not added to the DOM, until the user actually want's this dialog to be shown